### DISCUSSION TITLE
Multiple shop locations

### DISCUSSION THREAD

**Discussion by trouble2** [openstore-ecommerce/OpenStore#136] [Q&A]
Any ideas about this?
So we have an import csv with the stock for each product for each store.
So the shop has 3 physical locations, and there would be 3 records for each product in the csv.

- option 1 would be to add the stock levels of all 3 shops in the import where the productid is the same. Is that possible?
- option 2 would be to add three products, (or perhaps models ?) with their own stock level and 'do' something with that ?

**Comment by leedavi** [comment]
Do you mean a company with 3 portals (website), that share products?

Or just each product (or model) is sent from 3 different locations and then they update the stock?

It seem to me to have 1 product with 3 models?  But, I'm not full aware of what your situation is.

Can you explain a little more?

**Comment by trouble2** [comment]
Well, that might also be an option, to create a portal for each location/shop....
But I actually mean 1 portal/webshop...

**Reply by leedavi** [reply]
OK, I see you only want to update the stock qty, which is uploaded by the CSV file.  right?

In that case option1 should work.  And would probably be the easiest.  We've recently done something like this, but with an external system which send the update via an API, which is far more automatic.  But a CSV should be OK.

**Comment by trouble2** [comment]
So, you're saying you also got some kind of plugin that reads in the products through an api?
That would be very interesting, since they also have an external api with their products and stock etc. which updates every night....
Is this a plugin I can find somewhere?

**Reply by leedavi** [reply]
Sorry, this is closed source.  The client doesn't want it open source, for security.  

We are updating the order, not the product.  But the principle is the same.

My idea was to make this an API to move towards a mobile app.  It's fairly easy, I'm just trying to find the time to do it, rename and remove the clients bespoke code.  Then I'll release it open source. 

We have an update check by calling the clients API, with a secretkey.  To prove the data came from them.  This is ideal, but a security encryption secretkey could work as well. Opening up the API for update needs heavy security check to ensure it's not a hacker trying to update the system.
